frontend/components/data-table.js

```diff
@@ -0,0 +1,550 @@
+import Component from '../component.js';
+import Pagination from './pagination.js';
+import PaginationSelector from './pagination-selector.js';
+
+export default class DataTable extends Component {
+  constructor({ element, phones, columnConfig }) {
+    super({ element });
+
+    this._props = {
+      config: columnConfig,
+      defaultPhones: phones,
+      phonesLength: phones.length,
+      phones,
+    };
+
+    this._render();
+    this._init();
+  }
+
+  _init() {
+    const defaultPerPage = 5;
+    const { phonesLength } = this._props;
+
+    this._state = {
+      filterValue: '',
+      selectedFilter: 'all',
+      orderValue: 'up',
+      orderName: '',
+      mainCheckboxValue: false,
+      currentPage: 0,
+      totalItemsCount: this.getItemsCount(),
+      buttonsCount: Math.ceil(phonesLength / defaultPerPage),
+      itemsPerPage: defaultPerPage,
+      perPageOptions: [3, 5, 10, 15, 20],
+    };
+
+    let { defaultPhones } = this._props;
+
+    defaultPhones = this._addCheckFieldToItems(defaultPhones);
+    this.updateProps({ defaultPhones });
+
+    this._initFilter();
+    this._initPagination();
+    this._initPhonesCheckbox();
+    this._initFieldsEditing();
+
+    this._makeFiltering();
+  }
+
+  _initFilter() {
+    this._filterInput = this._getElement('find-input');
+
+    this.on('click', '[data-type="filter-selected"]', (event) => {
+      this._clearCheckButtonsStyle();
+
+      const btn = event.target.closest('[data-type="filter-selected"]');
+      btn.classList.add('header-button-active');
+      const selectedFilter = btn.dataset.value;
+
+      this._updateState({ selectedFilter });
+      this._resetCurrentPage();
+      this._makeFiltering();
+
+      const { itemsPerPage, buttonsCount } = this._state;
+      const { phonesLength } = this._props;
+
+      this._pagination.updateProps({
+        perPage: itemsPerPage,
+        phonesLength,
+        buttonsCount,
+      });
```

Comment:
the idea of separation a model from view is to move all the logic updating data to a render method
o `_updateView` in out case
So you only call the `_updateState` here and move all the rest logic there